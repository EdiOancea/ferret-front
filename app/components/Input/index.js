import React from 'react';
import PropTypes from 'prop-types';
import TextInput from 'components/TextInput';
import FileInput from 'containers/FileInput';
import { Field, ErrorMessage } from 'formik';

const Input = ({ name, label, type, setFieldValue }) => {
  const renderFields = ({ field, form }) =>
    name === 'images' ? (
      <FileInput {...{ name, field, setFieldValue }} />
    ) : (
      <TextInput {...{ name, label, type, form, field }} />
    );

  return (
    <div>
      <Field name={name}>
        {({ field, form }) => renderFields({ field, form })}
      </Field>
      <ErrorMessage name={name} />
    </div>
  );
};

Input.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  setFieldValue: PropTypes.func.isRequired,
};

export default Input;
