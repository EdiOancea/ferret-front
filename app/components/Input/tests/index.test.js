import React from 'react';
import { shallow } from 'enzyme';

import Input from 'components/Input';

describe('<Input />', () => {
  it('should not regress', () => {
    const props = {
      name: 'testName',
      label: 'testLabel',
      type: 'testType',
    };
    const tree = shallow(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
