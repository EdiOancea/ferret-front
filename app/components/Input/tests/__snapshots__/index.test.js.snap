// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should not regress 1`] = `
<div>
  <FormikConnect(FieldInner)
    name="testName"
  >
    <Component />
  </FormikConnect(FieldInner)>
  <FormikConnect(ErrorMessageImpl)
    name="testName"
  />
</div>
`;
